Update `translation/translators.rst` (#1576)
python/devguide: documentation/translations/translating.rst

===========
Translating
===========

.. highlight::  rest

There are several documentation translations already
in production; others are works in progress. To get started read your

> repositories contributing guide, which is generally the ``README``

Reviewer: ```suggestion
repository's contributing guide, which is generally the ``README``
```

   * - Arabic (ar)
     - Abdur-Rahmaan Janhangeer (:github-user:`Abdur-rahmaanJ`)
     - :github:`GitHub <Abdur-rahmaanJ/python-docs-ar>`
   * - `Bengali (bn_IN) <https://docs.python.org/bn-in/>`__
     - Kushal Das (:github-user:`kushaldas`)
     - :github:`GitHub <python/python-docs-bn-in>`
   * - `French (fr) <https://docs.python.org/fr/>`__
     - Julien Palard (:github-user:`JulienPalard`)
     - `AFPy/python-docs-fr <https://git.afpy.org/AFPy/python-docs-fr/>`_

>        :github:`Mirror <python/python-docs-fr>`

Reviewer 2: This is correct. French translation happens at AFPy's Gitea and are simply mirrored to GitHub.

Instead of "GitHub" or "AFPy", there could be "repo" or "repository" text which links to the proper translation repository (for all rows in the table). Please add a comma before "Mirror"

     - :github:`GitHub <sanketgarade/python-doc-mr>`
   * - Lithuanian (lt)
     - Albertas Gimbutas (:github-user:`albertas`, `email <mailto:albertasgim@gmail.com>`__)
     - `Original announcement <https://mail.python.org/pipermail/doc-sig/2019-July/004138.html>`__
   * - Persian (fa)
     - Alireza Shabani (:github-user:`revisto`)
     - :github:`GitHub <revisto/python-docs-fa>`
   * - `Polish (pl) <https://docs.python.org/pl/>`__
     - Maciej Olko (:github-user:`m-aciek`)
     - :github:`GitHub <python/python-docs-pl>`,
       `Transifex <tx_>`_,
       `Original announcement <https://mail.python.org/pipermail/doc-sig/2019-April/004106.html>`__
   * - Portuguese (pt)
     - Gustavo Toffo
     -
   * - `Brazilian Portuguese (pt-br) <https://docs.python.org/pt-br/>`__
     - | Rafael Fontenelle (:github-user:`rffontenelle`),
       | Marco Rougeth (:github-user:`rougeth`)
     - :github:`GitHub <python/python-docs-pt-br>`,

>        `Wiki <https://python.org.br/traducao/>`__,

Reviewer 2: Do you mean "Guide" instead of "Wiki" ?

>        `Wiki <https://python.org.br/traducao/>`__,

Reviewer 2: Sure, I agree it is a better word.

   * - `Turkish (tr) <https://docs.python.org/tr/>`__
     - Ege Akman (:github-user:`egeakman`)
     - :github:`GitHub <python/python-docs-tr>`,
       `RTD <https://python-docs-tr.readthedocs.io/>`__ XXX Why keep this with python build? Ask Ege
   * - `Ukrainian (uk) <https://docs.python.org/uk/>`__
     - Dmytro Kazanzhy (:github-user:`kazanzhy`, `email <mailto:dkazanzhy@gmail.com>`__)
     - :github:`GitHub <python/python-docs-uk>`,
       `Transifex <tx_>`_

How to get help
===============

If there is already a repository for your language team (there may be links to
Telegrams/Discords in the ``README``), join and introduce
yourself, your fellow translators will be more than happy to help! XXX sounds off
General discussions about translations occur on the Python Docs Discord
`#translations channel <https://discord.gg/h3qDwgyzga>`_, `translation
mailing list <translation_ml_>`_, and the `translations subsection <_discourse>`_

> of the Python Discourse.

Reviewer 2: Since some may not use a proprietary tool like Discourse, maybe keeping a instant messaging room would be interesting. But I'd suggest to migrate to Matrix instead of Libera Chat considering that there is no bridge between them anymore, and Matrix has great adoption.

> of the Python Discourse.

Reviewer: @rffontenelle I have never used Matrix.  Is there discussion happening there?  What place in particular, and how active is it?

I appreciate that some people prefer non-proprietary tools, but we are asking them to use GitHub.  It's more important that people can talk to each other, even on proprietary tools.

Also, isn't Discourse open?  https://github.com/discourse/discourse

> of the Python Discourse.

Reviewer 3: @rffontenelle An interesting suggestion about Matrix. Python has over the years had a number of mailing lists, IRC channels (thank you @nedbat for moderating), Zulip, Discourse, Discord, and issues on GitHub/bugs.python.org. I'm not against trying another service but that would be a wider decision and also need input from the Infra team (Ee and others). 

Perhaps for now, we can recommend that translators use discuss.python.org (Discourse), Discord, GitHub issues, and translation mailing list for discussion.

These may be lines from Monty Python skits or other references, they are, however,
sometimes obscure and lose meaning when translated, so do not feel pressured
to get them perfect.

.. XXX Ask EB can we do this QOL improvement?/Reword too

Don't translate keywords or names, including variable, function, class, argument,
and attribute names. An example of a translated codeblock from the `tutorial <https://docs.python.org/3/tutorial/controlflow.html#keyword-arguments>`_
is provided below:

.. code-block:: python

   def cheeseshop(kind, *arguments, **keywords):
       print("-- Czy jest może", kind, "?")
       print("-- Przykro mi, nie mamy już sera", kind)
       for arg in arguments:
           print(arg)
       print("-" * 40)
       for kw in keywords:
           print(kw, ":", keywords[kw])

Translation quality
-------------------

Translators should be proficient in both English and the language they are
translating to. Translators should aim for a similar level of quality as that
of the English documentation.

Avoid relying solely on machine translation. These tools can be useful to speed up
work, but often produce inaccurate or misleading results and should be reviewed.

> .. XXX ask EB about permission for this section or maybe https://github.com/python/peps/pull/4296/ to ref

Reviewer: TBH, we don't need the Editorial Board to decide things like this.  The current translators are in the best position to make recommendations about how to use all tools, including machine translation.  My understanding is that many people feel that current LLMs excel at translation.

Vocabulary
----------

The documentation is full of technical terms, some are common in general
programming and have translations, whereas others are specific to Python.
Translations should keep the translations of these terms consistent, which is
done with glossaries/translation memory.

.. XXX Reword above
.. XXX note that in coordinating.rst

Some general guidelines for deciding on a translation:

.. XXX todo (a lot)

- Use existing community conventions over inventing new terms.
- Use hybrid form
- Use English for common terms ... maybe hard to desc this case
- Use your best judgment
- Use other translations for reference as to what to do
- Be careful to not translate names
- Record in glossary to help peers!!!!!!!!!!!!!!!!!

Dialects
--------

Some translation receive contributions from people of many different dialects,
understandably the language will differ, it is recommended however that
translators try to keep files/large sections consistent.

Transifex
=========

.. XXX maybe add quickstart?

.. important::

>    There are many translations in the `python transifex <tx_>`_, some of which,

Reviewer 2: ```suggestion
   There are many translations in the `python-doc organization on Transifex <tx_>`_, some of which,
```

   however, not used anymore or do not have a coordinator, please confirm this
   is not the case before you begin translating.

Translations on Transifex are carried out via a web interface, similar to Weblate.
For further information about Transifex, and our guides on getting started, see
our `documentation <https://python-docs-transifex-automation.readthedocs.io/new-translators.html>`_.

> .. XXX Move stuff here, discuss w Rafael?

Reviewer 2: Looks good, as it can be the entry point for any newcomer in translation for Transifex as well.

The Placeholders page is more like a documentation for maintainers of the python-doc on Transifex, and to publicize what custom placeholders are set since it is not transparent for others users when browsing the organization's settings.

If you need help from a Transifex administrator, open an issue on the
`tracker <https://github.com/python-docs-translations/transifex-automations/issues>`_.
.. XXX explain that it is not same as coordinator

Translation FAQ
===============

Which version of the Python documentation should I work on?
-----------------------------------------------------------

You should work on the latest branch available for you translation, the translations

> should then be propagated by the languages coordinator.

Reviewer 2: "available for you translation" doesn't seem correct. Did you mean "available for you" or "available for translation"?

Also consider using "coordinator(s)" since you suggested before the language team to have one or more coordinators.

Author: I will reword it.

Good point, I will use `coordination team` I think.

The coordinator for my language is inactive, what do I do?
----------------------------------------------------------

If you would like to coordinate, follow the (necessary) steps outlined in
:doc:`coordinating` and open an issue in the
`EB issue tracker <https://github.com/python/editorial-board/issues>`_.

> .. XXX Ask EB if this is ok

Reviewer 3: I would have them open an issue on the Dev Guide requesting to be added to the translation chart. I think it is fine to have the person @-mention the EB `@editorial-board` in the devguide issue to review the PR. I don't think we need a PR on the editorial board repo each time. Thoughts, @nedbat?

   * - `Turkish (tr) <https://docs.python.org/tr/>`__
     - Ege Akman (:github-user:`egeakman`)
     - :github:`GitHub <python/python-docs-tr>`,
       `RTD <https://python-docs-tr.readthedocs.io/>`__ XXX Why keep this with python build? Ask Ege
   * - `Ukrainian (uk) <https://docs.python.org/uk/>`__
     - Dmytro Kazanzhy (:github-user:`kazanzhy`, `email <mailto:dkazanzhy@gmail.com>`__)
     - :github:`GitHub <python/python-docs-uk>`,
       `Transifex <tx_>`_

How to get help
===============

If there is already a repository for your language team (there may be links to
Telegrams/Discords in the ``README``), join and introduce

> yourself, your fellow translators will be more than happy to help! XXX sounds off

Reviewer: ```suggestion
yourself. Your fellow translators will be more than happy to help!
```

     - :github:`GitHub <python/python-docs-uk>`,
       `Transifex <tx_>`_

How to get help
===============

If there is already a repository for your language team (there may be links to
Telegrams/Discords in the ``README``), join and introduce
yourself. Your fellow translators will be more than happy to help!
General discussions about translations occur on the Python Docs Discord
`#translations channel <https://discord.gg/h3qDwgyzga>`_, `translation
mailing list <translation_ml_>`_, and the `translations subsection <_discourse>`_
of the Python Discourse.

Style Guide
===========

Before translating, you should familiarize yourself with the general
documentation :doc:`style guide<../style-guide>`. Some translation-specific
guidelines are explained below.

Translate, don't rewrite
------------------------

Try to stay as close as possible to the original text. Focus on translating its

> meaning rather than rephrasing or rewriting it.

Author: Sometimes it is necessary, yes, but more often it is not. I think we should have this guideline, so that new translators don't rewrite everything. The guidelines should be strict IMO, as whatever we write practice will still differentiate.

How about we add "in the best possible way," and then it reads as: limit your re-writing, rather than an outright stop to it.

> meaning rather than rephrasing or rewriting it.

Reviewer 4: I agree with @m-aciek, we don't want rigid translations, we want people to understand what the docs are trying to get across, and rewriting may be the best way.

Gender neutrality
-----------------

Many languages use grammatical gender. When possible and natural, prefer
gender-neutral or inclusive forms. Aim to reflect the inclusive tone of
the English documentation.

Roles and links
---------------

The Python docs contain many roles (``:role:`target```)
to other parts of the documentation.
Leave reStructuredText roles such as ``:func:`print``` or ``:ref:`some-section``` in
place, even if they contain section titles, because it will break the link.

> If alternate text (``:role:`text <target>``` is provided, it can be translated.

Author: I agree, though this seems risky, since syntax translators have to ensure the syntax is correct, and that people do not translate text like ``` :class:`list` ```. I will add it for now, but we may have to revisit this in the future.

Links (```text <target>`_``) should be handled similarly. If possible, the target
should be updated to match the language.

.. seealso::
   :doc:`../markup`

Code examples
-------------

Translate values in code examples, that is string literals, and comments.
Don't translate keywords or names, including variable, function, class, argument,
and attribute names. An example of a translated codeblock from the `tutorial <https://docs.python.org/3/tutorial/controlflow.html#keyword-arguments>`_
is provided below:

.. code-block:: python

   def cheeseshop(kind, *arguments, **keywords):
       print("-- Czy jest może", kind, "?")
       print("-- Przykro mi, nie mamy już sera", kind)
       for arg in arguments:
           print(arg)
       print("-" * 40)
       for kw in keywords:
           print(kw, ":", keywords[kw])

Translation quality
-------------------

Translators should be proficient in both English and the language they are
translating to. Translators should aim for a similar level of quality as that

> of the English documentation.

Reviewer 5: Not sure if any translation project is using two-phased process of translation: translation/review, but I can imagine that in such process the first-phase translator could be only intermediate in both English and the second language.

I think we should leave the requirements of the languages proficiency up to translation projects. For example, for some intermediate English should be enough, especially at early stages of translation.

I suggest us dropping the first sentence and leaving only the one about the quality.

Reviewer 5: I still would argue that it should be left up to projects decision. With current translation tools, medium English could be enough especially for some low-traffic projects. I think that sometimes not ideal translation can encourage new contributors. I would like us to be inclusive here, inviting everyone with a good will. I think that polishing ones English on such a project should be fine for some teams.

Please note that I'm not suggesting to lower the target quality, rather to allow the process to be more iterative and not limit the contributors because of their education.

> Avoid relying solely on machine translation. These tools can be useful to speed up
> work, but often produce inaccurate or misleading results and should be reviewed
> by a human.

Reviewer 5: ```suggestion
Please do not rely solely on machine translation. These tools can be useful to speed up
work, but often produce inaccurate or misleading results and should be reviewed
by a human.
```
In my opinion “avoid” is too soft in this context.

Author: Good point, I went with "do not rely".

Terminology
-----------

The documentation is full of technical terms, some are common in general
programming and have translations, whereas others are specific to Python
and previous translations are not available (remember to check your languages

> glossary). Translations should keep the translations of these terms consistent,

Reviewer 2: Did you mean "translators" in here?

```suggestion
glossary). Translators should keep the translations of these terms consistent,
```

> The documentation is full of technical terms, some are common in general
> programming and have translations, whereas others are specific to Python
> and previous translations are not available (remember to check your languages
> glossary). Translation teams should keep the translations of these terms
> consistent, which is done with glossaries and or translation memory.

Reviewer 5: ```suggestion
The documentation is full of technical terms, some are common in general
programming and have translations, whereas others are specific to Python
and previous translations are not available.
Translation teams should keep the translations of these terms
consistent, which is done with glossaries.
```
I think the translation memory has marginal use for that, the glossaries are definitely the solution for consistent terminology, used in both offline and online tools.

Some general guidelines for deciding on a translation:

> - Avoid neologism when possible, use existing community conventions over
>   inventing new terms.

Reviewer 5: ```suggestion
- Use existing community conventions over inventing new terms.
```
Neologism in this context can be understood contrary to the intention:
> In [linguistics](https://en.wikipedia.org/wiki/Linguistics), a neologism ([/niˈɒləˌdʒɪzəm/](https://en.wikipedia.org/wiki/Help:IPA/English); also known as a coinage) is any newly formed word, term, or phrase that has achieved popular or institutional recognition and is becoming accepted into mainstream language.[[1]](https://en.wikipedia.org/wiki/Neologism#cite_note-1) Most definitively, a word can be considered a neologism once it is published in a dictionary.[[2]](https://en.wikipedia.org/wiki/Neologism#cite_note-2)

- You can use a hybrid English form if users are generally familiar
  with the English word.
- For common terms, the English word may be best.
- Use other translations as a reference as to what they did for the word.
- Be careful to not translate names.

> - Use your best judgment

Reviewer 5: ```suggestion
- Use your best judgment.
```

- When you translate a specific term, record in glossary to help fellow
  translators and ensure consistency.

Dialects
--------

Some translation receive contributions from people of several different dialects,
understandably the language will differ, it is recommended however that
translators try to keep files and sections consistent.

Transifex
=========

.. important::

>    There are many translations in the `python-doc organization on transifex <tx_>`_,

Reviewer 2: ```suggestion
   There are many translations in the `python-doc organization on Transifex <tx_>`_,
```

>    some of which, however, not used anymore or do not have a coordination team,

Reviewer 5: ```suggestion
   some of which, however, not used or do not have a coordination team,
```
only one is not used anymore, others were never pulled

> Transifex
> =========
> 
> .. important::
> 
>    There are many translations in the `python-doc organization on Transifex <tx_>`_,
>    some of which, however, not used anymore or do not have a coordination team,
>    please confirm this is not the case before you begin translating.
> 
> Translations on Transifex are carried out via a web interface, similar to Weblate.

Reviewer 5: ```suggestion
Translation platform
====================

.. important::

   There are many translations in the `python-doc organization on Transifex <tx_>`_,
   some of which, however, not used anymore or do not have a coordination team,
   please confirm this is not the case before you begin translating.

Some of the language projects use Transifex as a translation platform.
Translations on Transifex are carried out via a web interface, similar to Weblate.
```

Author: I think it's better to leave it so we can have a Weblate specific section down the line once that becomes available. I like your wording though, so I add it.

Reviewer 5: In my opinion a platform change should be all-projects-wide, we shouldn't support multiple platforms in a long term. I think this section could be switched to Weblate if/when we switch to it, and/or also describe the switching process during the transition phase.

>    There are many translations in the `python-doc organization on Transifex <tx_>`_,
>    some of which, however, not used anymore or do not have a coordination team,
>    please confirm this is not the case before you begin translating.
> 
> Translations on Transifex are carried out via a web interface, similar to Weblate.
> You should translate the **python-newest** project.

Reviewer 5: ```suggestion
   There are many translations in the ``python-doc`` organization on Transifex,
   some of which, however, not used anymore or do not have a coordination team,
   please confirm this is not the case before you begin translating.

Translations on Transifex are carried out via a web interface, similar to Weblate.
You should translate the `**python-newest** <tx_>`_ project.
```

Telegrams/Discords in the ``README``), join and introduce
yourself. Your fellow translators will be more than happy to help!
General discussions about translations occur on the Python Docs Discord
`#translations channel <https://discord.gg/h3qDwgyzga>`_, `translation
mailing list <translation_ml_>`_, and the `translations subsection <_discourse>`_
of the Python Discourse.

Style Guide
===========

Before translating, you should familiarize yourself with the general
documentation :doc:`style guide<../style-guide>`. Some translation-specific
guidelines are explained below.

Translate, don't rewrite
------------------------

Try to stay as close as possible to the original text. Focus on translating its
meaning in the best possible way, rather than rephrasing or rewriting it.

Gender neutrality
-----------------

Many languages use grammatical gender. When possible and natural, prefer
gender-neutral or inclusive forms. Aim to reflect the inclusive tone of
the English documentation.

Roles and links
---------------

The Python docs contain many roles (``:role:`target```)
to other parts of the documentation.

> Leave reStructuredText roles such as ``:func:`print``` or ``:ref:`some-section``` in
> place, even if they contain section titles, because it will break the link.
> If alternate text (``:role:`text <target>``` is provided, it can be translated.
> You can also introduce alternate text for translation if the target is not a
> name or term.

Reviewer 5: ```suggestion
Do not translate reStructuredText role targets (``:func:`print``` or
``:ref:`name <target>```), because it will break the link.
Alternate text (as in``:role:`text <target>```) often should be translated.
You can also introduce alternate text for translation, especially when a target term is used in a colloquial English context.
```
suggestion

Author: Thanks, but I reworded it slightly differently. I think we do not need to list "especially when a" but rather "especially not when a".

should be updated to match the language.

.. seealso::
   :doc:`../markup`

Translation quality
-------------------

Translators should be proficient in both English and the language they are
translating to. Translators should aim for a similar level of quality as that
of the English documentation.

Do not rely solely on machine translation. These tools can be useful to speed up
work, but often produce inaccurate or misleading results and should be reviewed
by a human.

Terminology
-----------

The documentation is full of technical terms, some are common in general
programming and have translations, whereas others are specific to Python
and previous translations are not available.
Translation teams should keep the translations of these terms
consistent, which is done with glossaries.

Some general guidelines for deciding on a translation:

- Use existing community conventions over inventing new terms.
- You can use a hybrid English form if users are generally familiar
  with the English word.
- For common terms, the English word may be best.
- Use other translations as a reference as to what they did for the word.
- Be careful to not translate names.
- Use your best judgment.

> - When you translate a specific term, record in glossary to help fellow

Reviewer 4: ```suggestion
- When you translate a specific term, record in a glossary to help fellow
```

Author: I clarified that it should be the shared glossary instead.

  translators and ensure consistency.

Dialects
--------

Some translation receive contributions from people of several different dialects,

> understandably the language will differ, it is recommended however that

Reviewer 4: ```suggestion
understandably the language will differ. It is recommended however that
```
